lib/template.js

```diff
@@ -23,7 +28,12 @@ module.exports = function(config) {
     return input.toISOString().substring(0, 10);
   });
 
-  const sitemapSrc = config.sitemap.template || join(__dirname, '../sitemap.xml');
+  // if post.date or post.updated not exists, add default today date lastmod
+  env.addGlobal('today', () => {
+    return new Date().toISOString().substring(0, 10);
+  });
```

Comment:
https://hexo.io/docs/configuration#Date-Time-format

Hexo has `updated_option` to determine the behavior when `page.updated` not present.

Reply:
i dont know why, sometime `lastmod` is not present on sitemap after generate.
i think should give an fallback to current date (today).

Reply:
yesterday my google search console give me result sitemap format invalid because invalid date (i dont know, and that notifications disapeared today 6 hours after i add this fallback)

Reply:
oh i see, sometimes date parsing error because writing article on low devices. and made some function not working and indicated as invalid and return as NULL. thats why sometimes the `<lastmod>` not present in sitemap (this causes search engine to stop crawling the sitemap)

Reply:
basically this is necessary to avoid error from hexo function (either late return or fail parsing as null)

Reply:
can i mark resolved this convertisation ?

Reply:
> i dont know why, sometime `lastmod` is not present on sitemap after generate. i think should give an fallback to current date (today).

No. It will cause `lastmod` to be changed every time you generate your site, which is actually pretty bad to SEO. Search engine like Google and Yandex will think you are trying to cheat the system to crawl your site more often and stop indexing your site entirely.

Reply:
but the lastmod sometime is not present in <url><loc></....</priority>. then should i cancel this pull ?

any ideas to avoid nullable post.updated or post.date ?

maybe run on this event https://hexo.io/api/events.html#exit , but hexo on exit have an bug hexojs/hexo#4822

Reply:
> but the lastmod sometime is not present in `<url><loc></....</priority>`. then should i cancel this pull ?
> 
> any ideas to avoid nullable post.updated or post.date ?
> 
> maybe run on this event https://hexo.io/api/events.html#exit , but hexo on exit have an bug [hexojs/hexo#4822](https://github.com/hexojs/hexo/issues/4822)

I told you, Hexo has an option for that, to use existing `date` for `updated`, or use file last modified date as `updated`, or keep it `null` as it is. Do not create your own behavior that is not consistent with Hexo.

Reply:
this node function needed on hexo api. https://stackoverflow.com/questions/14031763/doing-a-cleanup-action-just-before-node-js-exits

Reply:
Strange conversation where @SukkaW told you how to fix the issue and you did not understand or care and continue talking about ideas that would not fix your issue

Reply:
He already told me how hexo work on post.updated variable. And i should cancel this pulls, because `Do not create your own behavior that is not consistent with Hexo.`
I saw on sitemap.xml on this repository would not written `<lastmod>` when post.updated or post.date not defined (no fallback) and he told me if i using my code it will overwritten lastmod Everytime i created post. 

all hexo admins do not provide convenience for users with low specs. and hexo uses 90% of resources when generating static files (1000 more posts), and causes some variables to be nullable. And no fixable solution here.

Reply:
https://github.com/hexojs/hexo-generator-sitemap/blob/b59601e3a3ad7849bcaa82a7a027eed948521741/sitemap.xml#L6

Reply:
Perangkat saya hanya memiliki prosesor I3 dan ram 8gb.
terkadang file hexo dirender atau terkadang tidak dibuat halaman html yang benar, begitupun dengan sitemap.

Reply:
If the admin answers NO, then what can I as a user do?

Reply:
now, i just have to writting my own plugin that suitable with my problems.

![image](https://user-images.githubusercontent.com/12471057/146849606-ee5971ee-f15e-44a7-b340-bf1cc017e936.png)

Reply:
no lodash, su*ks memory / cpu usages.

Reply:
why use an external library if it can be accessed directly. wasting memory resources.

![image](https://user-images.githubusercontent.com/12471057/146850062-2ec0b668-493c-46b4-a658-3f2bbb0d977c.png)

Reply:
and this plugin does not have ability to sphread sitemap if more than 5000 pages (including non-hexo files)